src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -288,22 +293,52 @@ public function walkSelectorSource(QOM\SelectorInterface $source)
     }
 
     /**
+     * @param QOM\JoinConditionInterface $right
+     *
+     * @return string the alias on the right side of a join
+     *
+     * @throws \BadMethodCallException if the provided JoinCondition has no valid way of getting the right selector
+     */
+    private function getRightJoinSelector(QOM\JoinConditionInterface $right)
+    {
+        if ($right instanceof QOM\ChildNodeJoinConditionInterface) {
+            return $right->getParentSelectorName();
+        } elseif ($right instanceof QOM\DescendantNodeJoinConditionInterface) {
+            return $right->getAncestorSelectorName();
+        } elseif (method_exists($right, 'getSelector2Name')) {
+            return $right->getSelector2Name();
```

Comment:
okay, we can't just add new interfaces for this. but you could do `$right instanceof QOM\SameNodeJoinConditionInterface || $right instanceof QOM\EquiJoinConditionInterface`. those are the only qom interfaces with the getSelector2Name. i would prefer that, the method_exists is probably slower and feels strange. the PHPCR interfaces extending QOM\JoinConditionInterface are those 4. to hit the error situation, somebody would need to write their own QOM class implementing JoinConditionInterface directly instead of any of the 4 specific interfaces.